R/reactable.R

```diff
@@ -597,12 +597,15 @@ reactable <- function(data, columns = NULL, columnGroups = NULL,
 
   htmlwidgets::createWidget(
     name = "reactable",
-    reactR::reactMarkup(component),
+    x = reactR::reactMarkup(component),
     width = width,
     height = height,
     package = "reactable",
     dependencies = dependencies,
-    elementId = elementId
+    elementId = elementId,
+    preRenderHook = function(instance) {
```

Comment:
Is there a specific reason to set the theme in `preRenderHook` instead of `reactable()`? I guess I'm just generally wondering what `preRenderHook` is used for, since there's not much documentation about it out there.

Reply:
> Is there a specific reason to set the theme in preRenderHook instead of reactable()?

Yes, `preRenderHook` gets called at print-time, just before the widget is serialized into JSON. By calling `bs_current_theme()` at print-time, we can have some guarantee of knowing when a `{bslib}` theme has been provided to a page layout function like `fluidPage()`:

```r
library(shiny)

ui <- fluidPage(
  theme = bslib::bs_theme(),
  verbatimTextOutput("txt")
)

server <- function(input, output, session) {
  output$txt <- renderPrint(bslib::bs_current_theme())
}

shinyApp(ui, server)
```

Reply:
Ahh gotcha, that makes sense.